index.js

```diff
@@ -112,102 +113,94 @@ function cachingFetch (uri, _opts) {
         throw err
       } else {
         // Missing cache entry, or mode is default (if stale), reload, no-store
-        return remoteFetch(uri, opts)
+        return remoteFetch(req.url, opts)
       }
     })
   } else {
     return remoteFetch(uri, opts)
   }
 }
 
-// https://tools.ietf.org/html/rfc7234#section-4.2
-function isStale (res) {
-  if (!res) { return null }
-  const ctrl = res.headers.get('Cache-Control') || ''
-  if (ctrl.match(/must-revalidate/i)) {
-    return true
-  }
-  if (ctrl.match(/immutable/i)) {
-    return false
+function adaptHeaders (headers) {
+  const newHs = {}
+  for (let k of headers.keys()) {
```

Comment:
Why use `let` here? I’m wondering because you seem to prefer `const` elsewhere in the code.